[RPC profiling] Add test to ensure using record_function works for RPC
functions

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#43657 [RPC profiling] Add test to ensure using record_function works for RPC
functions**
functions**
functions**
functions**

We didn't have a test that ensures functions ran over RPC that are being profiled can use `with record_function()` to profile specific blocks in the function execution. This is useful if the user wants certain information about specific blocks in the function ran over RPC composed of many torch ops and some custom logic, for example.

Currently, this will not work if the function is TorchScripted since `with record_function()` is not torchscriptable yet. We can add support for this in future PRs so that torchscript RPC functions can also be profiled like this.

Differential Revision: [D23355215](https://our.internmc.facebook.com/intern/diff/D23355215/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D23355215/)!

rohan-varma committed:
[RPC profiling] Add test to ensure using record_function works for RPC
functions

We didn't have a test that ensures functions ran over RPC that are being profiled can use `with record_function()` to profile specific blocks in the function execution. This is useful if the user wants certain information about specific blocks in the function ran over RPC composed of many torch ops and some custom logic, for example.

Currently, this will not work if the function is TorchScripted since `with record_function()` is not torchscriptable yet. We can add support for this in future PRs so that torchscript RPC functions can also be profiled like this.

Differential Revision: [D23355215](https://our.internmc.facebook.com/intern/diff/D23355215/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D23355215/)!

[ghstack-poisoned]

mrshenli commented on torch/testing/_internal/distributed/rpc/rpc_test.py:
does the order of the two contexts matter here?

rohan-varma commented on torch/testing/_internal/distributed/rpc/rpc_test.py:
It only matters because it can control the type of tensor created by `at::cpp:custom_type_hack::create` but since we're not testing that here and it doesn't affect the behavior of record function, the order should not matter.

rohan-varma committed:
Update on "[RPC profiling] Add test to ensure using record_function works for RPC
functions"

functions**
functions**

We didn't have a test that ensures functions ran over RPC that are being profiled can use `with record_function()` to profile specific blocks in the function execution. This is useful if the user wants certain information about specific blocks in the function ran over RPC composed of many torch ops and some custom logic, for example.

Currently, this will not work if the function is TorchScripted since `with record_function()` is not torchscriptable yet. We can add support for this in future PRs so that torchscript RPC functions can also be profiled like this.

Differential Revision: [D23355215](https://our.internmc.facebook.com/intern/diff/D23355215/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D23355215/)!

[ghstack-poisoned]

rohan-varma committed:
Update on "[RPC profiling] Add test to ensure using record_function works for RPC
functions"

functions**
functions**
functions**

We didn't have a test that ensures functions ran over RPC that are being profiled can use `with record_function()` to profile specific blocks in the function execution. This is useful if the user wants certain information about specific blocks in the function ran over RPC composed of many torch ops and some custom logic, for example.

Currently, this will not work if the function is TorchScripted since `with record_function()` is not torchscriptable yet. We can add support for this in future PRs so that torchscript RPC functions can also be profiled like this.

Differential Revision: [D23355215](https://our.internmc.facebook.com/intern/diff/D23355215/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D23355215/)!

[ghstack-poisoned]
